.github/workflows/compilation_on_android_ubuntu.yml

```diff
@@ -323,6 +323,17 @@ jobs:
           sudo tar -xzf wabt-1.0.24-*.tar.gz
           sudo mv wabt-1.0.24 wabt
 
+      - name: build wasi-libc (needed for wasi-threads)
+        run: |
+          git clone --branch wasi-sdk-17 https://github.com/WebAssembly/wasi-libc
```

Comment:
Branch wasi-sdk-17 isn't found in the wasi-libc repo, seems that main branch is OK.

Reply:
It's a tag https://github.com/WebAssembly/wasi-libc/tree/wasi-sdk-17. 

I'd prefer to pin the version/tag rather than using main. Unless we want to use the current latest commit from main and we checkout that specific one.

Reply:
I remember that the combination `wasi-sdk-16` + `main` from `wasi-libc` was failing on Ubuntu.

Reply:
OK. So shall we merge the PR now? @loganek, are there any comments from you? Thanks.

Reply:
I think so, to recap:
- we're using `wasi-sdk-16` because `wasi-sdk-17` would require a newer version of libc wrt the one on Ubuntu-20.04 used in the pipeline (the alternative would be to update the Ubuntu image)
- `--export=malloc` and `--export=free` are used here otherwise the pipeline would fail when executing the `wasi-threads` samples (the alternative would be to comment the assert in `samples/wasi-threads/wasm-apps/no_pthread.c`)